### ISSUE TITLE
Rename UnsafeCast to CastNeverSucceed

### ISSUE THREAD

**Issue by xouabita** [arturbosch/detekt#2153]
Related to https://github.com/arturbosch/detekt/issues/2131:

Looks like after https://github.com/arturbosch/detekt/pull/1879 this rule doesn't only apply on [unsafe casts](https://kotlinlang.org/docs/reference/typecasts.html#unsafe-cast-operator) but also to [safe casts](https://kotlinlang.org/docs/reference/typecasts.html#safe-nullable-cast-operator) that [cannot succeed](https://github.com/arturbosch/detekt/blob/2af3fe204df949f6fabe76267306e4af9caa74c9/detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnsafeCastSpec.kt#L27-L33).

eg:
```kotlin
fun test(s: String) {
    println((s as? Int) ?: 0)
}
```

is now reported.

Is it possible to rename the rule to `CastNeverSucceed` to disambiguate it with unsafe casts ?

Also, the documentation seems outdated: [this](https://github.com/arturbosch/detekt/blob/2af3fe204df949f6fabe76267306e4af9caa74c9/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnsafeCast.kt#L19) is compliant with the [new behavior](https://github.com/arturbosch/detekt/blob/2af3fe204df949f6fabe76267306e4af9caa74c9/detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnsafeCastSpec.kt#L36-L39).

Happy to open a pull request if you agree 🙂

**Comment by schalkms** [comment]
Thanks for taking the time to write an improvement issue. Definitely, the docs should be updated. I’m not favoring a rename here, since this would be a breaking change for v1.3.0 and I’m not sure if it really provides much value. However, I think it would be better to write a better more thorough description for this rule.

Please go ahead! We really appreciate your help in that regard.

**Comment by schalkms** [comment]
Closed by 8f56648
